org.metaborg.spoofax.meta.core/src/main/java/org/metaborg/spoofax/meta/core/pluto/build/StrIncrFrontEnd.java

```diff
@@ -0,0 +1,161 @@
+        private static final long serialVersionUID = 3808911543715367986L;
+
+        public final String moduleName;
+        public final Map<String, File> generatedFiles;
+
+        public Output(String moduleName, Map<String, File> generatedFiles) {
+            this.moduleName = moduleName;
+            this.generatedFiles = generatedFiles;
+        }
+    }
+
+    // Just a type alias
+    public static class BuildRequest extends
+        build.pluto.builder.BuildRequest<Input, Output, StrIncrFrontEnd, SpoofaxBuilderFactory<Input, Output, StrIncrFrontEnd>> {
+        private static final long serialVersionUID = -1299552527869341531L;
+
+        public BuildRequest(SpoofaxBuilderFactory<Input, Output, StrIncrFrontEnd> factory, Input input) {
+            super(factory, input);
+        }
+    }
+
+    public static SpoofaxBuilderFactory<Input, Output, StrIncrFrontEnd> factory =
+        SpoofaxBuilderFactoryFactory.of(StrIncrFrontEnd.class, Input.class);
+
+    public static BuildRequest request(Input input) {
+        return new BuildRequest(factory, input);
+    }
+
+    public static Origin origin(Input input) {
+        return Origin.from(request(input));
+    }
+
+    public StrIncrFrontEnd(Input input) {
+        super(input);
+    }
+
+    @Override protected Output build(Input input) throws Throwable {
+        requireBuild(input.origin);
+
+        require(input.inputFile);
+
+        IStrategoTerm result = callStrategoCompileBuilder(context.resourceService().resolve(input.inputFile));
+
+        String moduleName = Tools.javaStringAt(result, 0);
+        IStrategoList strategyList = Tools.listAt(result, 1);
+        Map<String, File> generatedFiles = new HashMap<>();
+
+        for(IStrategoTerm strategyTerm : strategyList) {
+            String strategy = Tools.asJavaString(strategyTerm);
+            File file = context.toFile(paths.strSepCompStrategyFile(moduleName, strategy));
+            generatedFiles.put(strategy, file);
+            provide(file);
+        }
+
+        provide(context.toFile(paths.strSepCompBoilerplateFile(moduleName)));
+
+        setState(State.finished(true));
+        return new Output(moduleName, generatedFiles);
+    }
+
+    @Override protected String description(Input input) {
+        return "Compile Stratego to separate strategy ast files";
+    }
+
+    @Override public File persistentPath(Input input) {
+        final Path rel = FileCommands.getRelativePath(context.baseDir, input.inputFile);
+        final String relname = rel.toString().replace(File.separatorChar, '_');
+        return context.depPath("str_sep_front." + relname + ".dep");
+    }
+
+    public IStrategoTerm callStrategoCompileBuilder(FileObject resource) throws IOException {
+        final BuildInputBuilder inputBuilder = new BuildInputBuilder(context.project());
+        ILanguageImpl strategoLangImpl = context.languageService().getLanguage(STRATEGO_LANG_NAME).activeImpl();
```

Comment:
@Gohla I get a `NullPointerException` here. Is this not the way to the Stratego language impl?